### DISCUSSION TITLE
How to install on EKS with Calico

### DISCUSSION THREAD

**Discussion by druchoo** [longhorn/longhorn#6335] [Q&A]
Hello, I'm trying Longhorn (v1.5.0) on EKS 1.24 with Calico using the official Helm Chart with all default values. longhorn-manager pods throw the following error and restart:

```
time="2023-07-15T20:44:00Z" level=fatal msg="Error starting manager: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9502/v1/webhook/validaton?timeout=10s\": Address is not allowed"
```

Editing the `longhorn-manager` DaemonSet and setting `hostNetwork: true` resolves the issue. I see no mention of this in the docs or any issues that I could find. It's also not possible to set hostNetwork from the Chart. 

Is this the correct approach? Are there any issues with running longhorn-manager with host networking?

Thanks

**Comment by mantissahz** [comment]
Hi, @druchoo, 

It seems to be a limitation of EKS.
https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks#install-eks-with-calico-networking

>**NOTE**
Calico networking cannot currently be installed on the EKS control plane nodes. As a result the control plane nodes will not be able to initiate network connections to Calico pods. (This is a general limitation of EKS's custom networking support, not specific to Calico.) As a workaround, trusted pods that require control plane nodes to connect to them, such as those implementing admission controller webhooks, can include hostNetwork:true in their [pod spec](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec). See the Kubernetes API pod spec definition for more information on this setting.

Agree to have a document to explain why this happened and the workaround.

@longhorn/qa Have we tested the installation on the EKS with the Calico?

**Reply by yangchiu** [reply]
> Have we tested the installation on the EKS with the Calico?

No, we only tested EKS default CNI.

**Reply by c3y1huang** [reply]
https://github.com/longhorn/longhorn/issues/6426
